pkg/ipamclaimswebhook/podmutator.go

```diff
@@ -233,54 +302,117 @@ func ensureIPAMClaimRefAtNetworkSelectionElements(ctx context.Context,
 	return hasChangedNetworkSelectionElements, nil
 }
 
-func findNewPrimaryNetworkIPAMClaimName(ctx context.Context,
-	cli client.Client, pod *corev1.Pod, vmName string) (string, error) {
+func findPrimaryUDNLogicalNetworkAttachment(
+	ctx context.Context,
+	vmi *virtv1.VirtualMachineInstance,
+	pluginConfig *config.RelevantConfig,
+) (*virtv1.Interface, error) {
 	log := logf.FromContext(ctx)
-	if pod.Annotations[config.OVNPrimaryNetworkIPAMClaimAnnotation] != "" {
-		return "", nil
+
+	// TODO: once OVN-K lands this feature, we can remove uncomment this code
+	//if !pluginConfig.AllowPersistentIPs {
+	//	return nil, nil
+	//}
```

Comment:
@qinqon please remind me: you're already removing this in your https://github.com/ovn-kubernetes/ovn-kubernetes/pull/5321 PR, aren't you ?

Reply:
I bumped nad-client to remove limitation of iprequest + ipamclaimref